charmcraft/config.py

```diff
@@ -150,7 +171,18 @@ def format_pydantic_errors(errors):
         formatted_loc = format_pydantic_error_location(error["loc"])
         formatted_msg = format_pydantic_error_message(error["msg"])
 
-        combined.append(f"- {formatted_msg} in field {formatted_loc!r}")
+        if formatted_msg == "field required":
+            field_name, location = printable_field_location_split(formatted_loc)
+            combined.append(
+                f"- field {field_name} required for {location} configuration"
+            )
+        elif formatted_msg == "extra fields not permitted":
+            field_name, location = printable_field_location_split(formatted_loc)
+            combined.append(
+                f"- extra field {field_name} not permitted for {location} configuration"
```

Comment:
"required for" and "permitted for", or (taking in consideration that the problematic field happens inside a config key, "required in" and "permitted in"?

Reply:
I actually did it both ways and kinda flip-flopped.  Take your pick and I'll update it if desired :)